### PR TITLE
Ttva 146 sap xml doc

### PR DESCRIPTION
**Pull request by danjacob-anders**
https://andersinno.atlassian.net/jira/software/c/projects/TTVA/boards/11?modal=detail&selectedIssue=TTVA-146

This PR includes the following changes:

- model changes to Reservations to handle invoice approval/generation workflow
- API changes to manage "invoice requested" state changes for admin (see https://github.com/andersinno/varaamo/pull/122 for corresponding changes in Varaamo)
- SAP XML invoice generation 

The XML files for now are written to file by a Django command. It is expected that this command will be updated to also dispatch the invoices via FTP to TietoEvry (or other provider) in a subsequent ticket. Some values have been added to the XML document where possible, otherwise dummy placeholder values have been used.

### CODE REVIEW

File: payments/templates/payments/sap/sales_order.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        {% for item in order.items %}
+            <Item>
+            <Description>{{ item.description }}</Description>
+            <ProfitCenter>{{ item.cost_center_code }}</ProfitCenter>
+            <InternalOrder>000000027070</InternalOrder>
+            <Material>{{ item.material }}</Material>
+            <PriceCondition>ZMYH</PriceCondition>
+            <Currency>{{ item.currency }}</Currency>
+            <Quantity>{{ item.quantity }}</Quantity>
+            <UnitPrice>{{ item.unit_price|unlocalize }}</UnitPrice>
```
Comment (danjacob-anders) [inline payments/templates/payments/sap/sales_order.xml:37]: Decimal format should be in international standard e.g. "12.50" not "12,50" for "twelve euro 50 cent".

File: payments/templates/payments/sap/sales_order.xml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        <InterfaceID>XXX</InterfaceID>
+        <Text>
+        <TextRow>{{ billing_period }}</TextRow>
+        </Text>
+        <Items>
+        {% for item in order.items %}
+            <Item>
+            <Description>{{ item.description }}</Description>
+            <ProfitCenter>{{ item.cost_center_code }}</ProfitCenter>
+            <InternalOrder>000000027070</InternalOrder>
```
Comment (danjacob-anders) [inline payments/templates/payments/sap/sales_order.xml]: Dummy values, to be fixed/removed

File: payments/templates/payments/sap/sales_order.xml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            <Street>{{ order.address.street }}</Street>
+            <Town>{{ order.address.town }}</Town>
+            <PostCode>{{ order.address.postcode }}</PostCode>
+        </address>
+        </Customer>
+        <BillingDate>{{ order.billing_date }}</BillingDate>
+        <ServiceRenderDate>{% now "Y-d-m" %}</ServiceRenderDate>
+        <BillNumber>20186245</BillNumber>
+        <YourReference>{{ order.reference }}</YourReference>
+        <SalesOrganisation>1111</SalesOrganisation>
```
Comment (danjacob-anders) [inline payments/templates/payments/sap/sales_order.xml]: Dummy values, to be fixed/removed

File: payments/sap_invoices.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        .prefetch_related(
+            "order__order_lines",
+            "order__order_lines__product",
+        )
+        .order_by("invoice_approved_at")
+    )
+
+    yield reservations
+
+    reservations.update(invoice_generated_at=timezone.now())
```
Comment (jopesy) [inline payments/sap_invoices.py]: I think this function should only be used for getting the eligible reservations, and then marking them as generated (`invoice_generated_at`) should happen elsewhere, e.g. inside `generate_sales_order` or in the management command. Of course the tests would need to be adjusted then too.

Comment (danjacob-anders) [reply payments/sap_invoices.py]: Yes, that's fine, that probably makes more sense.

**Review by jopesy — APPROVED** [review]
Top-quality PR once again. Looks great!

**Pull request merged**